packages/@uppy/aws-s3-multipart/src/createSignedURL.js

```diff
@@ -0,0 +1,98 @@
+function createCanonicalRequest ({
+  method = 'PUT',
+  CanonicalUri = '/',
+  CanonicalQueryString = '',
+  SignedHeaders,
+  HashedPayload,
+}) {
+  const headerKeys = Object.keys(SignedHeaders).map(k => k.toLowerCase()).sort()
+  return [
+    method,
+    CanonicalUri,
+    CanonicalQueryString,
+    ...headerKeys.map(k => `${k}:${SignedHeaders[k]}`),
+    '',
+    headerKeys.join(';'),
+    HashedPayload,
+  ].join('\n')
+}
+
+const { subtle } = globalThis.crypto
+const ec = new TextEncoder()
+const algorithm = { name: 'HMAC', hash: 'SHA-256' }
+
+async function digest (data) {
+  return subtle.digest(algorithm.hash, ec.encode(data))
+}
+
+async function generateHmacKey (secret) {
+  return subtle.importKey('raw', typeof secret === 'string' ? ec.encode(secret) : secret, algorithm, false, ['sign'])
+}
+
+function arrayBufferToHexString (arrayBuffer) {
+  const byteArray = new Uint8Array(arrayBuffer)
+  let hexString = ''
+  for (let i = 0; i < byteArray.length; i++) {
+    hexString += byteArray[i].toString(16).padStart(2, '0')
+  }
+  return hexString
+}
+
+async function hash (key, data) {
+  return subtle.sign(algorithm, await generateHmacKey(key), ec.encode(data))
+}
+
+export default async function createSignedURL ({
```

Comment:
Needs docs and a test

Reply:
25kiB is certainly not worth it, my implementation is more tailored to our needs. I don't think publishing as a standalone npm package would make sense 🤔 it's meant to be an Uppy-only thing, if folks want to use it elsewhere, they're welcome to fork it.

Reply:
Jest is _again_ putting itself on the way of perfectly valid code test 😒  I'm not sure what to do of it, the test passes with `node:test`, fails with Jest because it messes up the ESM dependencies.

Reply:
That sucks. Do what you think is best then 👍